### ISSUE TITLE
Missing witness field when parsing response from Esplora

### ISSUE THREAD

**Issue by cryptoquick** [bitcoindevkit/bdk#570]
**Describe the bug**  

My program has recently started to panic when parsing Esplora responses. Not sure what changed or happened.

I'm provided with this specific error:

```
panicked at 'called `Result::unwrap()` on an `Err` value: Esplora(Reqwest(reqwest::Error { kind: Decode, source: Error("missing field `witness`", line: 1, column: 1038) }))', src/lib.rs:211:14
```

**To Reproduce**  

This is from calling unwrap on the result from the wallet constructor.

```rust
    let wallet = Wallet::new(
        &descriptor,
        Some(&change_descriptor),
        *NETWORK.read().unwrap(),
        MemoryDatabase::default(),
        blockchain,
    )
    .await?;
```

**Expected behavior**  

N/A

**Build environment**  
 - BDK tag/commit: 0.16.1
 - OS+version: Arch Linux
 - Rust/Cargo version: 1.59.0
 - Rust/Cargo target: wasm32-unknown-unknown

**Additional context**  
 

https://github.com/diba-io/bitmask-core/blob/b4db810d90c1e7622e0440769162e08cb2aecc71/src/lib.rs#L211

![Screenshot from 2022-03-15 12-57-35](https://user-images.githubusercontent.com/285690/158452779-feddceaa-9a22-4cd2-af30-6a104bdbfdee.png)

**Comment by LLFourn** [comment]
This is probably because esplora does not return a `witness` field for non-segwit inputs. Looks like we don't have a test for that :S.

[EDIT]

Or maybe coinbase inputs?

Are you willing to share the outpoint this is spending from (it looks like testnet)?

**Comment by LLFourn** [comment]
@cryptoquick It was that pre-segwit inputs were broken with esplora. I'm not sure for how long. Perhaps since #461.

Please confirm if #571 fixes it.

**Comment by cryptoquick** [comment]
@LLFourn That works perfectly! And really good job on the test. There were a couple of changes I had to make to our wallet when upgrading to the latest version of bdk, but after I made those, it all works great now. I'm very grateful for how quickly you were able to help me out, too!

You might appreciate just glancing over the changes I made in order to use your fix:
https://github.com/diba-io/bitmask-core/pull/12/files
